overlord/snapstate/snapmgr.go

```diff
@@ -176,23 +186,46 @@ func (snapst *SnapState) SetTryMode(active bool) {
 	}
 }
 
+func autherForUserID(st *state.State, userID int) (store.Authenticator, error) {
+	var auther store.Authenticator
+	if userID > 0 {
+		user, err := auth.User(st, userID)
+		if err != nil {
+			return nil, err
+		}
+		auther = user.Authenticator()
+	}
+	return auther, nil
+}
+
+func updateInfo(st *state.State, name, channel string, userID int, flags Flags) (*StoreUpdateInfo, error) {
+	auther, err := autherForUserID(st, userID)
+	if err != nil {
+		return nil, err
+	}
+	devmode := flags&DevMode > 0
+	remoteInfo, err := Store().Snap(name, channel, devmode, auther)
+	if err != nil {
+		return nil, err
+	}
+	return &StoreUpdateInfo{
+		DownloadURL:     remoteInfo.DownloadURL,
+		AnonDownloadURL: remoteInfo.AnonDownloadURL,
+		Confinement:     string(remoteInfo.Confinement),
+		SideInfo:        remoteInfo.SideInfo,
+	}, nil
+}
+
 // Manager returns a new snap manager.
 func Manager(s *state.State) (*SnapManager, error) {
 	runner := state.NewTaskRunner(s)
 
-	storeID := ""
-	// TODO: set the store-id here from the model information
-	if cand := os.Getenv("UBUNTU_STORE_ID"); cand != "" {
-		storeID = cand
-	}
-	store := store.NewUbuntuStoreSnapRepository(nil, storeID)
 	// TODO: if needed we could also put the store on the state using
 	// the Cache mechanism and an accessor function
```

Comment:
^ this is my was my original place if we had to access the store without being able to access the manager